paddle/framework/operator.cc

```diff
@@ -99,5 +99,48 @@ std::string OperatorBase::DebugString() const {
   return ss.str();
 }
 
+template <>
+const Variable* OperatorContext::Input<Variable>(int index) const {
+  return scope_->GetVariable(op_.inputs_[index]);
+}
+
+template <>
+Variable* OperatorContext::Output<Variable>(int index) const {
+  return scope_->GetVariable(op_.outputs_[index]);
+}
+
+template <>
+const Variable* OperatorContext::Input<Variable>(
+    const std::string& name) const {
+  return scope_->GetVariable(op_.Input(name));
+}
+
+template <>
+Variable* OperatorContext::Output(const std::string& name) const {
+  return scope_->GetVariable(op_.Output(name));
+}
+
+template <>
+const std::vector<const Variable*> OperatorContext::MultiInput(
+    const std::string& name) const {
+  auto names = op_.Inputs(name);
+  std::vector<const Variable*> res;
+  std::transform(
+      names.begin(), names.end(), res.begin(),
```

Comment:
`std::back_inserter(res)` ? because res is empty, transform cannot copy data to res container.